### PR TITLE
[READY] Prisoners can now latejoin with DROPPODS

### LINKED ISSUE #52340: Contractor return pod isnt working
## Round ID:
[142316](https://scrubby.melonmesa.com/round/142316)

## Reproduction:
Get kidnapped by a contractor and sent to ninja jail. Wait out the timer, once you are supposed to go back you instead get sent to the gem room due to being nullspaced. The items supposed to be sent on the same pod does show up on the turf.
See /datum/syndicate_contract/proc/returnVictim(var/mob/living/M)
The items that are sent home with you on the same pod somehow shows up at the destination.

Possibly related to pod rework @MrDoomBringer

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

prisoner amounts got increases to 3 starting/4 max, that was previously only a thing on ice box but now every map except pubby has a large prison (in delta's case it's pr'd but not merged yet)
pubby still has 2 slots
prisoners can now latejoin, they get droppodded onto one of the prisoner spawn spots and theres a small radio announcement to the sec team
fixes #52340

video: https://streamable.com/qudav2

## Why It's Good For The Game

some more prisoner fun and fixes bugs

## Changelog
:cl:
add: prisoners can now late join through the SKY
fix: contractor stuff no longer teleports to nullspace
/:cl:

### CODE REVIEW

File: code/modules/cargo/supplypod.dm [hunk 1]
```diff
@@ -503,7 +503,7 @@
 			var/atom/movable/O = single_order
 			O.forceMove(pod)
 	for (var/mob/living/M in pod) //If there are any mobs in the supplypod, we want to set their view to the pod_landingzone. This is so that they can see where they are about to land
-		M.reset_perspective(src)
```
Comment (MrDoomBringer) [inline code/modules/cargo/supplypod.dm:506]: this isn't the right way to fix the issue, please hold

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

**Comment by MrDoomBringer** [comment]
otherwise this is a delightful PR

**Comment by MrDoomBringer** [comment]
one day every job will join like this

**Comment by Timberpoes** [comment]
What's the fallback for if the spawn area is not safe for spawning?

The shuttle is a pseudo-guaranteed safe space protected from griefing. Do you have anything in place to guard against people being drop-podded into a destroyed perma brig/plasma flooded permabrig/vented to space permabrig etc?

**Comment by Fikou** [comment]
> Do you have anything in place to guard against people being drop-podded into a destroyed perma brig/plasma flooded permabrig/vented to space permabrig etc?

no

**Comment by MrDoomBringer** [comment]
when you're a prisoner your rights to life are forefeit

**Comment by RaveRadbury** [comment]
#48819
image

**Comment by Timberpoes** [comment]
It less about the role and more about latejoining as prisoner and spawning in to your death.

The same reason that the shuttle is protected - Players should be given a chance to get their bearings and sort their shit out before they're mercilessly slaughtered. It's a protection for the sake of the players - They have absolutely no way to know the state of the station from the main menu.

**Comment by optimumtact** [comment]
There is no need for prisoners to be a late join role.

**Comment by ATH1909** [comment]
Remove prisoners from the role blacklist on the HoP's console so that HoPs can open up more prisoner slots.

**Comment by evilcyborgmrt** [comment]
There is no need for prisoners to not be a late join role

**Comment by optimumtact** [comment]
actually it's pretty clear there is, which is why these shitty half assed attempts to fix how to get the prisoners into the prison via some IC means

File: code/modules/cargo/supplypod.dm [hunk 2]
```diff
@@ -80,8 +81,35 @@
 	landingDelay = 20 //Very speedy!
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
 
-/obj/structure/closet/supplypod/Initialize()
+/obj/structure/closet/supplypod/securitypod
+	style = STYLE_SECURITY
+	bluespace = TRUE
+	explosionSize = list(0,0,0,1)
+	var/obj/item/radio/Radio
```
Comment (bobbahbrown) [inline code/modules/cargo/supplypod.dm:88]: you should lowercase this var and also document it as required by documentation standards

File: code/modules/cargo/supplypod.dm [hunk 3]
```diff
@@ -80,8 +81,35 @@
...
+
+/obj/structure/closet/supplypod/securitypod/Initialize()
+	. = ..()
+	Radio = new /obj/item/radio(src)
+	Radio.listening = 0
```
Comment (bobbahbrown) [inline code/modules/cargo/supplypod.dm:93]: unless this is not a boolean var

```suggestion
	Radio.listening = FALSE
```

File: code/modules/cargo/supplypod.dm [hunk 4]
```diff
@@ -80,8 +81,35 @@
...
+
+/obj/structure/closet/supplypod/securitypod/Destroy()
+	QDEL_NULL(Radio)
+	return ..()
+
+/obj/structure/closet/supplypod/securitypod/preOpen()
+		Radio.set_frequency(FREQ_SECURITY)
+		Radio.talk_into(src, "SECURITY ALERT: New prisoner dropping into [get_area(src)].", FREQ_SECURITY)
```
Comment (bobbahbrown) [inline code/modules/cargo/supplypod.dm:101]: can you double check that set_frequency is necessary? i suspect that's for continued transmissions or listening, you probably only need to do the talk_into based on the fact it takes a frequency but im just guessing

File: code/modules/cargo/supplypod.dm [hunk 5]
```diff
@@ -80,8 +81,35 @@
...
+		QDEL_NULL(Radio)
+		..()
+
+/obj/structure/closet/supplypod/proc/QueueAnnounce(mob, rank)
```
Comment (bobbahbrown) [inline code/modules/cargo/supplypod.dm:105]: needs dmdoc

Comment (bobbahbrown) [reply code/modules/cargo/supplypod.dm:105]: also make it lowercase (queue_announce)

**Comment by CRITAWAKETS** [comment]
make it so that if the droppod lands on a space tile instead they are transported to a random space z-level or the gulag

File: code/modules/cargo/supplypod.dm (hunk 4 above)
Comment (Fikou) [reply code/modules/cargo/supplypod.dm]: it is

**Comment by Fikou** [comment]
> actually it's pretty clear there is, which is why these shitty half assed attempts to fix how to get the prisoners into the prison via some IC means

what is the reason? i dont see why they cant be latejoin

**Comment by optimumtact** [comment]
read the PR again carefully, paying attention to the comment by Timberpoes please

**Comment by Fikou** [comment]
ok i guess ill try to make it a shuttle or something

**Comment by Dennok** [comment]
Maybe spawning a whole prisoner room?

**Comment by MrDoomBringer** [comment]
What? This is really not a difficult problem
Just make the pod pick a turf that is pressurized
if there's none, make the pod appear somewhere else in security, and give it a lock that needs to be opened by security personnel
and if all of security is dead, just let em appear in arrivals in cuffs or something

**Comment by MrDoomBringer** [comment]
could probably even skip that middle step if you wanted to keep things simple

**Comment by optimumtact** [comment]
as I said to fikou in discord, do it properly
prison shuttle
management console for warden to disable/enable accepting new prisoners
maximum limit of prisoners that can't be controlled by warden or HoP (to avoid flooding the permabrig with too many prisoners)

**Comment by MrDoomBringer** [comment]
well yeah that would probably be better

**Pull request closed without merging**